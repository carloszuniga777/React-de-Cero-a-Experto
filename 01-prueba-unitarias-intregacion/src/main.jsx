import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './styles.css'
import { CounterApp } from './CounterApp'
// import { FirstApp } from './FirstApp'


createRoot(document.getElementById('root')).render(
  <StrictMode>
     <CounterApp value={ 20 }/>
     {/* <FirstApp title='Hola, soy vegeta'/> */}
  </StrictMode>,
)
