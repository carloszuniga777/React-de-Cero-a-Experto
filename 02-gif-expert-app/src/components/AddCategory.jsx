import { useState } from "react"
import PropTypes from 'prop-types'


export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState('')


  const onSubmit = (event)=>{
     event.preventDefault()
     
     if(inputValue.trim().length <= 0) return
     
     onNewCategory(inputValue.trim())

     setinputValue('')
  } 
  
  
  return (
    <form onSubmit={onSubmit}>
        <input 
            type="text" 
            placeholder="Buscar gift"
            value={inputValue}   
            onChange={e => setinputValue(e.target.value)} 
        />
    </form>
  )
}


// Definir los tipos de las props
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired, // Define que es una funci√≥n requerida
};