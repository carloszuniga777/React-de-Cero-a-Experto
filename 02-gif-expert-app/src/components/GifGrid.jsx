
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from "prop-types"
import { GiftItem } from "./GiftItem"


export const GifGrid = ({category}) => {
  const { images, isLoading} = useFetchGifs(category)

  return (
    <>
       <h3>{category}</h3>
       {
          isLoading && (<div className="">Cargando...</div>)
       }
       <div className="card-grid">
              {        
                images.map(image=>(
                  <GiftItem key={image.id} {...image}/>
                ))
              }
       </div>
    </>
  )
}


// Definir los tipos de las props
GifGrid.propTypes = {
  category: PropTypes.string.isRequired, // `category` es un string obligatorio
};