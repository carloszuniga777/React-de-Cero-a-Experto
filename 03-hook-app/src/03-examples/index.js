

export * from './LoadingMessage';
export * from './MultipleCustomHooks';
