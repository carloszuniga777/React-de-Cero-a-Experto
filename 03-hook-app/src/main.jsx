import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './index.css'
import { BrowserRouter } from "react-router";



//import { FormWithCustomHook } from './FormWithCustomHook/FormWithCustomHook';

//UseState
//import { CounterApp } from './01-useState/CounterApp';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';

//useEffect
// import { SimpleForm } from './02-useEffect/SimpleForm';

//Custom Hook
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
 
//Multiples hooks
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';

//useRef
// import { FocusScreen } from './04-useRef/FocusScreen';

//memo
// import { Memorize } from './06-memos/Memorize';

//useMemo
// import { MemoHook } from './06-memos/MemoHook';

//callbackHook
// import { CallbackHook } from './06-memos/CallbackHook';
//import { Padre } from './07-tarea-memo/Padre';

//useReducer
//import { TodoApp } from './08-useReducer/TodoApp'

//useContext
import { MainApp } from './09-UseContext/MainApp';


createRoot(document.getElementById('root')).render(
  
  <BrowserRouter>
      <StrictMode>
          {/* <FormWithCustomHook/> */}
          {/* <CounterApp/> */}
          {/* <CounterWithCustomHook/> */}
          {/* <SimpleForm />  */}
          {/* <FormWithCustomHook/>   */}
          {/* <MultipleCustomHooks />  */}
          {/* <FocusScreen />  */}
          
          {/* <Memorize />  */}
          {/* <MemoHook />  */}
          
          {/* <CallbackHook />  */}
          {/* <Padre /> */}

        {/* <TodoApp /> */}

        <MainApp/>

      </StrictMode>
  </BrowserRouter>
)

