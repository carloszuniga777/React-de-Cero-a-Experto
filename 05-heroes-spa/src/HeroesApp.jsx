import { AuthProvider } from "./auth"
import { AppRouter, AppRouterNew } from "./router"  // eslint-disable-line



export const HeroesApp = () => {
  return (
    <AuthProvider>
       {/** <AppRouter/> */}    {/**Descomentar el BrowserRouter del archivo main.jsx */}
        
         <AppRouterNew/>            {/**Comentar el BrowserRouter del archivo main.jsx*/} 
    </AuthProvider>
  )
}
