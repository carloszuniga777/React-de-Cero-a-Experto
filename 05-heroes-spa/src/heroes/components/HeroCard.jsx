import { Link } from "react-router"

const CharactersByHero = ({alter_ego, characters})=>{
    return(
        <>
            {
                alter_ego !== characters && ( <p className="card-text">{alter_ego}</p>)
            }
        </>
    )
}


export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters
 }) => {

    const heroImageUrl = `/heroes/${ id }.jpg`;

  return (
    <div className="col animate__animated animate__fadeIn">
        <div className="card">
                <div className="row no-gutters">

                         {/**Imagen */}
                        <div className="col-4">
                            <img className='card-img'src={heroImageUrl} alt={superhero} />
                        </div>

                        <div className="col-8">
                            <h5 className="card-title">
                                { superhero }
                            </h5>

                            <CharactersByHero alter_ego={alter_ego} characters={characters} />

                            <p className="card-text">
                               <small className="text-muted">{characters}</small> 
                            </p>

                            <Link to={`/hero/${id}`}>
                                Mas..
                            </Link>
                        </div>
                </div>
        </div>
    </div>
  )
}
