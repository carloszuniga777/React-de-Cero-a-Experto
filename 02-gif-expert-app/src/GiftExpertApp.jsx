import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GiftExpertApp = () => {
 const [categories, setCategories] = useState([])



   /*La mejor manera de pasar el set del usestate al componente hijo, 
     es creando una funcion en el componente padre que haga referencia al set y pasarla al componente hijo */
  const onAddCategory = (newCategory)=>{
    
    //console.log(newCategory)
   
    if(categories.includes(newCategory)) return

    setCategories(cat => [newCategory, ...cat])
  }  




  return (
    <>
        <h1>GifExportApp</h1>
       
        <AddCategory onNewCategory={ value => onAddCategory(value)}/>
       
        {
            categories.map((category)=>(
                <GifGrid key={category} category={category}/>
            ))
        }
        
    </>
  )
}
