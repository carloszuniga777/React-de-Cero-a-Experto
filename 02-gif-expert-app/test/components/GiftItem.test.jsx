

import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components"
import { expect } from "vitest"




describe('pruebas en GifItem', ()=>{

    const title = 'Lol'
    const url = 'htpps://google.com.hn'
    let container_

    beforeEach(()=>{
      const { container } = render(<GiftItem title={title} url={url}/>)
      container_ = container 
    })



    test('deberia hacer match con el snaptshot', ()=>{
       expect( container_ ).toMatchSnapshot()
    })



    test('debe de mostrar la imagen con el URL y el ALT indicado', ()=>{
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)

    })


    test('debe de mostrar el tituli en el componente', ()=>{
        expect(screen.getByText(title)).toBeTruthy()
    })




})