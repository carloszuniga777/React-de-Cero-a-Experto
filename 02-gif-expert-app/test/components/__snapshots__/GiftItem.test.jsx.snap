// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pruebas en GifItem > deberia hacer match con el snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Lol"
      src="htpps://google.com.hn"
    />
    <p>
      Lol
    </p>
  </div>
</div>
`;
